src/microceph.py

```diff
@@ -338,6 +384,53 @@ def set_pool_size(pools: str, size: int):
     _run_cmd(cmd)
 
 
+@retry(wait=wait_fixed(5), stop=stop_after_attempt(10))
+def list_mgr_modules() -> dict:
+    """Returns a python dict of mgr modules.
+
+    available keys:
+       1. disabled_modules
+       2. always_on_modules
+       3. enabled_modules
+    """
+    cmd = ["ceph", "mgr", "module", "ls", "--format", "json"]
+    return json.loads(_run_cmd(cmd=cmd))
+
+
+def enable_mgr_module(module: str):
+    """Enable requested ceph mgr module."""
+    disabled_modules = [
+        module_info["name"] for module_info in list_mgr_modules()["disabled_modules"]
+    ]
+    if module not in disabled_modules:
+        logger.info("nothing to do, %s module is not disabled", module)
+        return
+
+    cmd = ["ceph", "mgr", "module", "enable", module]
+    _run_cmd(cmd=cmd)
+
+
+def disable_mgr_module(module: str):
+    """Disable requested ceph mgr module."""
+    enabled_modules = list_mgr_modules()["enabled_modules"]
+    if module not in enabled_modules:
+        logger.info("nothing to do, %s module is not enabled or is always on", module)
+        return
+
+    cmd = ["ceph", "mgr", "module", "disable", module]
+    _run_cmd(cmd=cmd)
+
+
+def enable_ceph_monitoring():
+    """Enable Monitoring for ceph cluster."""
+    enable_mgr_module("prometheus")
+
+
+def disable_ceph_monitoring():
+    """Disable Monitoring for ceph cluster."""
+    disable_mgr_module("prometheus")
```

Comment:
Wonder if having the extra disable_ceph_monitoring() really is worthwhile, could just call disable_mgr_module("prometheus"). Same w/ enable_ceph_monitoring() above

Reply:
as discussed, prometheus module is the basic of ceph monitoring, but there can also be the `ceph-exporter` enablement bits, configurations for perf metrics in the prometheus module etc. Hence monitoring enablement/disablement has be abstracted over the module enablement/disablement.